pkg/cmdimpl/add.go

```diff
@@ -349,134 +270,276 @@ func Add(args []string, allParams AddParamSet, deps *AddDependencies) error {
+		return errors.New("WeGO can not determine cluster status... exiting")
 	}
 
-	// Set up wego repository if required
-	wegoRepoName, err := fluxops.GetRepoName()
+	switch strings.ToUpper(params.AppConfigUrl) {
+	case string(ConfigTypeNone):
+		return addAppWithNoConfigRepo()
+	case string(ConfigTypeUserRepo):
+		return addAppWithConfigInUserRepo(ctx, deps.GitClient)
+	default:
+		return addAppWithConfigInExternalRepo(ctx, deps.GitClient)
+	}
+}
+
+func addAppWithNoConfigRepo() error {
+	// Source covers entire user repo
+	userSourceName := getUserRepoName()
+	userRepoSource, err := generateSource(userSourceName, params.Url, SourceTypeGit)
+	if err != nil {
+		return wrapError(err, "could not set up GitOps for user repository")
+	}
+	appYaml, err := generateAppYaml()
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create app.yaml for '%s'", params.Name))
+	}
+	// kustomize or helm referencing single user repo source
+	applicationGOAT, err := generateApplicationGoat(userSourceName)
 	if err != nil {
-		return wrapError(err, "could not get flux repo name")
+		return wrapError(err, fmt.Sprintf("could not create GitOps automation for '%s'", params.Name))
 	}
+	return applyToCluster(userRepoSource, appYaml, applicationGOAT)
+}
 
-	owner, err := getOwner()
+func addAppWithConfigInUserRepo(ctx context.Context, gitClient git.Git) error {
+	// Source covers entire user repo
+	userSourceName := getUserRepoName()
+	userRepoSource, err := generateSource(userSourceName, params.Url, SourceTypeGit)
 	if err != nil {
 		return err
 	}
-
-	fmt.Printf("Verifying %s repository exists...\n", wegoRepoName)
-	if _, err := gitproviders.RepositoryExists(wegoRepoName, owner); err != nil {
-		if errors.Is(err, gitprovider.ErrNotFound) {
-			fmt.Printf("Creating %s repository...\n", wegoRepoName)
-			if !params.DryRun {
-				if err := gitproviders.CreateRepository(wegoRepoName, owner, params.IsPrivate); err != nil {
-					return wrapError(err, "could not create repository")
-				}
-			}
-		} else {
-			return wrapError(err, "could not check repository exists")
-		}
+	targetKustomize, err := generateTargetKustomize(userSourceName, ".wego")
+	if err != nil {
+		return err
 	}
-
-	wegoRepoDir, err := ioutil.TempDir("", wegoRepoName)
+	appKustomize, err := generateAppKustomize(userSourceName, ".wego")
 	if err != nil {
 		return err
 	}
-	defer os.RemoveAll(wegoRepoDir)
+	if err := applyToCluster(userRepoSource, targetKustomize, appKustomize); err != nil {
+		return err
+	}
+	appYaml, err := generateAppYaml()
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create app.yaml for '%s'", params.Name))
+	}
+	applicationGoat, err := generateApplicationGoat(userSourceName)
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create GitOps automation for '%s'", params.Name))
+	}
+	if err := writeAppYaml(gitClient, appYaml, ".wego"); err != nil {
+		return err
+	}
+	if err := writeGoats(gitClient, ".wego", applicationGoat); err != nil {
```

Comment:
NIt: Since writeAppYaml and writeGoats are performing similar operations, I'd prefer to have the arguments in the same order.  i.e. base path then the manifests.